wasm/jit/jit_amd64_test.go

```diff
@@ -1404,6 +1402,58 @@ func TestAmd64Builder_handleCall(t *testing.T) {
 	})
 }
 
+func TestAmd64Builder_handleMemoryGrow(t *testing.T) {
+	builder := requireNewBuilder(t)
+
+	builder.initializeReservedRegisters()
+	// Emit memory.grow instructions.
+	builder.handleMemoryGrow()
+
+	// Compile.
+	code, err := builder.assemble()
+	require.NoError(t, err)
+
+	// Run code.
+	eng := newEngine()
+	mem := newMemoryInst()
+	jitcall(
+		uintptr(unsafe.Pointer(&code[0])),
+		uintptr(unsafe.Pointer(eng)),
+		uintptr(unsafe.Pointer(&mem.Buffer[0])),
+	)
+	// The function call must end with jitCallStatusCodeCallBuiltInFunction status.
+	require.Equal(t, jitCallStatusCodeCallBuiltInFunction, eng.jitCallStatusCode)
+	// Plus the index of builtin function must be that of memoryGrow.
+	require.Equal(t, int64(builtinFunctionIndexMemoryGrow), eng.functionCallIndex)
+}
+
+func TestAmd64Builder_handleMemorySize(t *testing.T) {
+	builder := requireNewBuilder(t)
+	builder.initializeReservedRegisters()
+	// Emit memory.size instructions.
+	builder.handleMemorySize()
+	// At this point, the size of memory should be pushed onto the stack.
```

Comment:
ex this is a good "what" comment because the assertion is a bit indirect. so this one helps a lot

Reply:
agreed yeah:D I've put too many comments including the obvious ones. Will keep each line more meaningful from now on!